misc/eni-trunking-validator/eni-trunking-validator.go

```diff
@@ -0,0 +1,245 @@
+		fmt.Fprintf(os.Stderr, "unable to get container metadata: %v\n", err)
+		os.Exit(1)
+	}
+
+	if err := verifyEndPoint(client, taskMetadataPath); err != nil {
+		fmt.Fprintf(os.Stderr, "unable to get task metadata: %v\n", err)
+		os.Exit(1)
+	}
+
+	if err := verifyEndPoint(client, containerStatsPath); err != nil {
+		fmt.Fprintf(os.Stderr, "unable to get container stats: %v\n", err)
+		os.Exit(1)
+	}
+
+	if err := verifyEndPoint(client, taskStatsPath); err != nil {
+		fmt.Fprintf(os.Stderr, "unable to get task stats: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func verifyIAMRole(client *http.Client) {
+	credentialsRelativeUri := os.Getenv("AWS_CONTAINER_CREDENTIALS_RELATIVE_URI")
+	if len(credentialsRelativeUri) == 0 {
+		fmt.Fprintf(os.Stderr, "AWS_CONTAINER_CREDENTIALS_RELATIVE_URI environment variable missing")
+		os.Exit(1)
+	}
+	fmt.Printf("IAM role found: %s\n", credentialsRelativeUri)
+
+	IAMRoleEndPoint := "http://169.254.170.2" + credentialsRelativeUri
+	_, err := metadataResponse(client, IAMRoleEndPoint)
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error pinging IAM role endpoint:%s\n", credentialsRelativeUri)
+		os.Exit(1)
+	}
+
+}
+
+func verifyHostName() {
+	cmd := exec.Command("hostname")
+	hostname, err := cmd.Output()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error executing `hostname` command: %v\n", err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("Hostname Found: %s\n", string(hostname))
+	_, errMatch := regexp.MatchString(`ip-[0-9]*-[0-9]*-[0-9]*-[0-9]*.*.internal`, string(hostname))
+	if errMatch != nil {
+		fmt.Fprintf(os.Stderr, "incorrect format hostname: %s\n", hostname)
+		os.Exit(1)
+	}
+}
+
+func verifyIMDSConnectivity(client *http.Client) {
+	_, err := metadataResponse(client, imdsEndpoint)
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error pinging imds endpoint:%s\n", imdsEndpoint)
+		os.Exit(1)
+	}
+}
+
+func verifySubnetGatewayEndpoint(client *http.Client) {
+	baseURL :=  imdsEndpoint + "network/interfaces/macs/"
+	body, err := metadataResponse(client, baseURL)
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error pinging endpoint: %s\n", baseURL)
+		os.Exit(1)
+	}
+	macAddress := strings.Split(string(body), "\n")[0]
+	fmt.Printf("Received mac address: %s \n", macAddress)
+	pingUrl := strings.Join([]string{baseURL, macAddress, "subnet-ipv4-cidr-block"}, "")
+	subnetCIDR, errSubnet := metadataResponse(client, pingUrl)
+	fmt.Printf("Received CIDR: %s\n", string(subnetCIDR))
+
+	if errSubnet != nil {
+		fmt.Fprintf(os.Stderr, "error pinging endpoint: %s\n", pingUrl)
+		os.Exit(1)
+	}
+
+	defaultSubnet := ComputeIPV4GatewayNetmask(string(subnetCIDR))
+	fmt.Printf("Subnet Found: %s\n", defaultSubnet)
+	pinger, errSubnetPing := ping.NewPinger(defaultSubnet)
+
+	if errSubnetPing != nil {
+		fmt.Fprintf(os.Stderr, "error pinging subnet: %s\n", errSubnetPing.Error())
+		os.Exit(1)
+	}
+
+	pinger.SetPrivileged(true) // ping will run with super-user privileges
+	pinger.Count = 10
+	pinger.Run() // blocks until finished
```

Comment:
can you specify a timeout for the pinger? see https://github.com/sparrc/go-ping/blob/master/ping.go#L109